.github/workflows/tests.yml

```diff
@@ -3,23 +3,63 @@ name: tests
 on:
   push:
   pull_request:
+    branches:
+      - main
+  workflow_dispatch:
   schedule:
-      - cron: '00 4 * * *'  # daily at 4AM
+      - cron: '0 0 * * 0'  # weekly
 
 jobs:
-  build:
+  unit-tests:
 
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.9"] # TODO: expand versions
```

Comment:
Before merging into main, is it well known which python version(s) community members are relying upon for compatibility?

Reply:
You're very right on this. We'll need to do a fairly rigorous delineation of min versions and CI for those. At the very least prior to issuing a new release. 

At this point there is very little community dependence and use (part of the rationale behind this PR).